dali/python/nvidia/dali/triton.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+pipeline_autoserialized = False
+
+
+def autoserialize(dali_pipeline):
+    """
+    Decorator, that marks a DALI pipeline (represented by :meth:`nvidia.dali.pipeline_def`) for
+    autoserialization in DALI Backend's model repository.
+
+    For details about the autoserialization feature, please refer to the DALI Backend documentation.
+
+    To properly autoserialize the DALI pipeline, the caller needs to run the Python script, that
+    contains the pipeline marked with ``autoserialize``, pass the target file name
+    (i.e. the path to the file, where the serialized model will be saved) and the magical command
+    ``autoserialize.me``. For example:
+
+        # identity.py
+        import nvidia.dali as dali
+
+        @dali.triton.autoserialize
+        @dali.pipeline_def(batch_size=3, num_threads=1, device_id=0)
+        def pipe():
```

Comment:
What about any arguments to the pipeline_def or the `pipe` function?

Reply:
This is a great idea. I believe the usage should be something like:
```python
@autoserialize(train_pipeline=False)
@pipeline_def
def pipe(train_pipeline=True):
   ...
```
Anyway, I'd like to make it rather a follow-up to this PR.

Reply:
I was thinking, that autoserialize could probably detect the arguments to `pipe` function, without the need to redeclare them in decorator and export (make available) them in the DALI backend as configuration? 
That would probably only work with builtins though.

Reply:
This won't go along with autoconfiguration of DALI Backend.

Yeah, we have lot's of `auto` stuff...

I mean, it will unless any of the `def pipe()` args does not have a default value. Then it won't be possible to infer the argument value anyhow. Treating `def pipe()` args as `kwargs` for the decorator resolves this issue and allows nicer error checking. Anyway, that's the reason I'd like to do this in the followup PR - we still need to discuss the solution.